Dockerfile

```diff
@@ -18,18 +25,18 @@ ARG BUILDTIME
 ARG GITCOMMIT
 ARG VERSION
 ARG IMAGE_REPO_PREFIX
-ENV GITCOMMIT=$GITCOMMIT VERSION=$VERSION BUILDTIME=$BUILDTIME IMAGE_REPO_PREFIX=$IMAGE_REPO_PREFIX
+ARG KUBERNETES_VERSION
+ENV GITCOMMIT=$GITCOMMIT VERSION=$VERSION BUILDTIME=$BUILDTIME IMAGE_REPO_PREFIX=$IMAGE_REPO_PREFIX KUBERNETES_VERSION=$KUBERNETES_VERSION
 ENV CGO_ENABLED=0
 RUN make bin/compose-controller bin/compose-controller.test e2e-binary bin/installer bin/api-server bin/api-server.test bin/e2e_benchmark
 RUN go get github.com/onsi/ginkgo/ginkgo
-RUN curl -LO https://storage.googleapis.com/kubernetes-release/release/v1.12.4/bin/linux/amd64/kubectl && \
+RUN curl -fLO https://storage.googleapis.com/kubernetes-release/release/v$KUBERNETES_VERSION/bin/linux/amd64/kubectl && \
```

Comment:
Curious why you added `-f`?

Reply:
I got $KUBERNETES_VERSION unset and curl returned some 404 status xml from google storage, resulting in a broken `kubectl` installation.
curl --fail will stop the build process on http != 2xx

Reply:
👍